tests/py2puml/inspection/test_inspectclass.py

```diff
@@ -0,0 +1,69 @@
+
+from importlib import import_module
+
+from py2puml.domain.umlclass import UmlClass, UmlAttribute
+from py2puml.domain.umlrelation import UmlRelation, RelType
+from py2puml.inspection.inspectmodule import inspect_module
+from py2puml.parsing.astvisitors import ConstructorVisitor
+from py2puml.parsing.moduleresolver import ModuleResolver
+from py2puml.parsing.parseclassconstructor import parse_class_constructor
+
+from tests.asserts.attribute import assert_attribute
+from tests.asserts.relation import assert_relation
+
+
+def test_inspect_module_should_find_static_and_instance_attributes():
+    domain_items_by_fqdn: Dict[str, UmlItem] = {}
+    domain_relations: List[UmlRelation] = []
+    inspect_module(
+        import_module('tests.modules.withconstructor'),
+        'tests.modules.withconstructor',
+        domain_items_by_fqdn, domain_relations
+    )
+
+    assert len(domain_items_by_fqdn) == 2, 'two classes were inspected'
+
+    # Coordinates UmlClass
+    coordinates_umlitem: UmlClass = domain_items_by_fqdn['tests.modules.withconstructor.Coordinates']
+    assert len(coordinates_umlitem.attributes) == 2, '2 attributes of Coordinates were inspected'
+    x_attribute, y_attribute = coordinates_umlitem.attributes
+    assert_attribute(x_attribute, 'x', 'float', False)
+    assert_attribute(y_attribute, 'y', 'float', False)
+
+    # Point UmlClass
+    point_umlitem: UmlClass = domain_items_by_fqdn['tests.modules.withconstructor.Point']
+    point_expected_attributes = {
+        'PI': ('float', True),
+        # 'origin': (None, True), # not annotated class variable -> not handled for now
+        'coordinates': ('Coordinates', False),
+        'day_unit': ('TimeUnit', False),
+        'hour_unit': ('TimeUnit', False),
+        'time_resolution': ('Tuple[str, TimeUnit]', False),
+        'x': ('int', False),
+        'y': ('str', False),
+        'x_unit': ('str', False),
+        'y_unit': ('str', False),
+        'z': (None, False),
+        'w': ('int', False),
+        'u': (None, False),
+        'v': (None, False),
+        'dates': ('List[date]', False),
+    }
+    assert len(point_umlitem.attributes) == len(point_expected_attributes), 'all Point attributes must be verified'
+    for attribute_name, (atrribute_type, attribute_staticity) in point_expected_attributes.items():
+        point_attribute: UmlAttribute = next((
+            attribute
+            for attribute in point_umlitem.attributes
+            if attribute.name == attribute_name
+        ), None)
```

Comment:
This part was not obvious at first read. Might this ease it?

```suggestion
        point_attribute: UmlAttribute = None
        for attribute in point_umlitem.attributes:
            if attribute.name == attribute_name
			    point_attribute = attribute
			    break
```

Reply:
ah, it is a question of habit and taste between for-break loops and comprehension/iterator. Actually, I kind of prefer iterators or comprehensions (I like these Python features): it is easy to forget to write the `break` instruction. Are you ok with keeping the iterator?

Reply:
Let's keep it as is to avoid adding nested scopes.